### PR TITLE
update MC7000 mapping info

### PR DESCRIPTION
**Pull request by toszlanyi**
This is the updated mapping information correlated to
https://github.com/mixxxdj/mixxx/pull/2949

### CODE REVIEW

File: source/hardware/controllers/denon_mc7000.rst [hunk 1]
```diff
@@ -79,13 +79,13 @@ Please check the \*.js mapping file for user variables to:
 
 -  activate NeedleDrop sensor while a track is playing (default: false)
 -  set the Pitch Fader ranges in % to toggle between them (default: 4,
-   6, 10, 16, 24)
+   6, 8, 10, 16, 24)
 -  Platter Ring LED mode: single LED on or off (default: 1). Can be
    switched with SHIFT + Deck button
 -  Vinyl Mode on or off at MIXXX start which also triggers the Platter
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:85]:
```suggestion
-  Vinyl Mode: on or off at Mixxx start which also triggers the Platter
```
We don't use all uppercase MIXXX, please also fix the other occurrences in this file.

File: source/hardware/controllers/denon_mc7000.rst [hunk 2]
```diff
@@ -298,27 +304,28 @@ Layout and functions
...
     create a Loop Out point at the current Location.
 
-43. **< / > Param 1/2:** These are currently used to beat jump backward
-    or forward by 8 beats. Press and hold Shift and then press these
-    buttons jumps 32 beats. *This section needs further development.*
-    **HELP APPRECIATED**
+43. **< / > Param 1/2:** These are currently mapped to add/remove rating
+    stars to the loaded track. Press and hold Shift and then press these
+    buttons to change track color in the library.
+    *This section needs further development.* **HELP APPRECIATED**
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:310]: What does this mean? Do you need help with documentation or with coding the mapping itself? 

If it's the the latter, does the mapping work as described here and just lacking additional features? I'd rather put a separate "Known issues" section at the bottom. If you really want to put this here, you should probably use a [Sphinx note](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html).

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thank you for taking care!

File: source/hardware/controllers/denon_mc7000.rst (hunk 2 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: Thanks! It was meant to get help for the mapping code but I changed the text and mention there are no 2nd and 3rd functions on the PAD buttons.

File: source/hardware/controllers/denon_mc7000.rst (hunk 1 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: done

File: source/hardware/controllers/denon_mc7000.rst [hunk 3]
```diff
@@ -192,7 +192,8 @@ Layout and functions
 
 23. **Deck:** Selects which deck in the software is controlled by that
     hardware deck. The left deck can control Deck 1 or 3; the right deck
-    can control Deck 2 or 4.
+    can control Deck 2 or 4. Press and hold Shift and then press this
+    button to trigger the Platter LEDs mode.
 
 24. **Shift:** Press and hold this button to access secondary functions
```
[the remark this answers is not in the document]
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:198]: Good to know. What's the benefit using :hwlabel:`SHIFT`: over **Shift:** ?

[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:198]: It enforces a consistent style when referring to labels on hardware. Some people wrote `Shift`, some wrote `SHIFT` or `[Shift]` or something like that. To the user, it becomes more readable as these are now emphasized by the stylesheet and they can now search for a button/knob/fader with that label on their controller even if the documentation does not have a schematic drawing with numbers.

Note that this only for *labeled* controls. Don't use it for Jog wheels when there isn't a label saying `JOG WHEEL` next to it.

[the remark this answers is not in the document]
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:198]: Thanks for explanation. Is there a guideline I can read through? Staying with the Jogwheel example it states **<SEARCH>** next to it using the Shift button. So would I need to bring that up a) separately b) just as comment for Jogwheel description or c) within the comment for Jogwheel like :hwlabel:`SEARCH`: ? So to extend it ... What about generally about shifted functions that are labelled on the controller ? How to define them in the manual? Cheers!

File: source/hardware/controllers/denon_mc7000.rst [hunk 4]
```diff
@@ -1,18 +1,12 @@
...
-Changing controls that deliver an unprecedented ability to instantly and
-harmonically match musical keys. The unit’s comprehensive, multi-input
-mixer also provides access to line/phono inputs from both authentic
-analogue (vinyl) and digital sources.
+The Denon MC7000 is a professional DJ controller which has got 4-channel capability and dual USB connections. These two USB audio interfaces enable two DJs to play together. This controller includes Denon’s high build quality and superior 24-bit audio reproduction, makes this suited to both mobile and club DJs.
+
+The dual 6" platters have a touch-capacitive design with rotational LED displays. The MC7000 has exclusive and dedicated Key Matching and Changing controls that deliver an unprecedented ability to instantly and harmonically match musical keys. The unit’s comprehensive, multi-input mixer also provides access to line/phono inputs from both authentic analogue (vinyl) and digital sources.
+
+Links
+~~~~~
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:9]: I don't think we have a separate "links" headline in the other controller docs.

```suggestion
```

File: source/hardware/controllers/denon_mc7000.rst [hunk 5]
```diff
@@ -31,39 +25,23 @@ Features
...
-      to
-      20.04 <https://ubuntu.com/tutorials/tutorial-upgrading-ubuntu-desktop#1-before-you-start>`__.
-   -  OpenSUSE Tumbleweed is currently on Kernel 5.6.x supporting the
-      MC7000 Audio Interface.
-   -  Manjaro Linux 19.0 and 20.0 feature Linux Kernel 5.4.x supporting
-      the MC7000 Audio Interface.
-
+   **5.4.21**. All kernel versions **5.6.x** and following are supporting the device. 
+   All recent Linux distributions are shipped with Audio support for the MC7000, like
+        -  Ubuntu 20.04 and 20.10
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:37]: unexpected indentation.

File: source/hardware/controllers/denon_mc7000.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: The document is re-formatted. Unfortunately when building the manual locally with 'make html' i get error messages that hwlabel could not be used (WARNING: Unknown interpreted text role "hwlabel".) so to review the finished site is not always as good.

Also with this commit now all tests fail. Anything I can do about it?

File: source/hardware/controllers/denon_mc7000.rst [hunk 6]
```diff
@@ -100,301 +78,377 @@ Layout and functions
    Denon MC7000 (schematic view)
 
 
-1.  **Input Selector:** Set this switch to the desired audio source from
-    this channel. Channel’s controls will only send MIDI information
-    when its input selector is set to USB.
+----
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:81]: Why all these lines? Just use a table if you want that:

```rst
==========  ==========
Col Head 1  Col Head 2
==========  ==========
Row 1       Some text for the second column of row 1.
Row 2       Some text for the second column of row 2.
==========  ==========
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: source/hardware/controllers/denon_mc7000.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst]: You need to need to install the dependencies (`pip install -r requirements.txt`) when you test locally. I suggest to do it in a [python virtual environment](https://docs.python.org/3/library/venv.html).

[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst]: After you push you can click on the deploy preview and check for errors.

[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst]: > Is there a guideline I can read through? Staying with the Jogwheel example it states **SEARCH** next to it using the Shift button. So would I need to bring that up a) separately b) just as comment for Jogwheel description or c) within the comment for Jogwheel like :hwlabel:`SEARCH`: ? So to extend it ... What about generally about shifted functions that are labelled on the controller ?

I think it makes sense to use

    :hwlabel:`SHIFT` + Jog Wheel

instead of:

    `hwlabel:`SEARCH`

I think that explicitly using SHIFT and the unshifted label makes it easier for the user to see what control you are talking about.

File: source/hardware/controllers/denon_mc7000.rst (hunk 4 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:9]: easy to remove it if you don't like.

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:81]: just wanted a separation of the single numbers. otherwise it is hard to read. Dont really need tables. would only complicate the things even more

File: source/hardware/controllers/denon_mc7000.rst (hunk 5 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:37]: I cannot see what is wrong...

Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:37]: In RST, you can just indent by a random number of spaces.

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:81]: Let's see how it looks once you fixed the deploy preview.

File: source/hardware/controllers/denon_mc7000.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: > You need to need to install the dependencies (`pip install -r requirements.txt`) when you test locally. I suggest to do it in a [python virtual environment](https://docs.python.org/3/library/venv.html).

I already have installed dependencies but now did again. "Requirement already satisfied:"
Still it doesnt like hwlabel.

`$ pip install -r requirements.txt
Requirement already satisfied: sphinx in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 1)) (3.2.1)
Requirement already satisfied: sphinx-intl in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 2)) (2.0.1)
Requirement already satisfied: graphviz in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 3)) (0.14.2)
Requirement already satisfied: markupsafe in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 4)) (1.1.1)
Requirement already satisfied: transifex-client in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 5)) (0.14.1)
Requirement already satisfied: sphinx-rtd-theme==0.4.3 in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 6)) (0.4.3)
Requirement already satisfied: sphinx-multiversion==0.2.1 in /home/tobias/.local/lib/python3.8/site-packages (from -r requirements.txt (line 7)) (0.2.1)
Requirement already satisfied: Jinja2>=2.3 in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (2.11.2)
Requirement already satisfied: babel>=1.3 in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (2.8.0)
Requirement already satisfied: packaging in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (20.4)
Requirement already satisfied: sphinxcontrib-devhelp in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.0.2)
Requirement already satisfied: sphinxcontrib-qthelp in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.0.3)
Requirement already satisfied: docutils>=0.12 in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (0.16)
Requirement already satisfied: alabaster<0.8,>=0.7 in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (0.7.12)
Requirement already satisfied: sphinxcontrib-serializinghtml in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.1.4)
Requirement already satisfied: Pygments>=2.0 in /usr/lib/python3/dist-packages (from sphinx->-r requirements.txt (line 1)) (2.3.1)
Requirement already satisfied: sphinxcontrib-applehelp in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.0.2)
Requirement already satisfied: snowballstemmer>=1.1 in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (2.0.0)
Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from sphinx->-r requirements.txt (line 1)) (45.2.0)
Requirement already satisfied: imagesize in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.2.0)
Requirement already satisfied: requests>=2.5.0 in /usr/lib/python3/dist-packages (from sphinx->-r requirements.txt (line 1)) (2.22.0)
Requirement already satisfied: sphinxcontrib-htmlhelp in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.0.3)
Requirement already satisfied: sphinxcontrib-jsmath in /home/tobias/.local/lib/python3.8/site-packages (from sphinx->-r requirements.txt (line 1)) (1.0.1)
Requirement already satisfied: click in /usr/lib/python3/dist-packages (from sphinx-intl->-r requirements.txt (line 2)) (7.0)
Requirement already satisfied: six<2.0.0 in /usr/lib/python3/dist-packages (from transifex-client->-r requirements.txt (line 5)) (1.14.0)
Requirement already satisfied: python-slugify<5.0.0 in /home/tobias/.local/lib/python3.8/site-packages (from transifex-client->-r requirements.txt (line 5)) (4.0.1)
Requirement already satisfied: gitpython<4.0.0 in /home/tobias/.local/lib/python3.8/site-packages (from transifex-client->-r requirements.txt (line 5)) (3.1.8)
Requirement already satisfied: urllib3<2.0.0,>=1.24.2 in /usr/lib/python3/dist-packages (from transifex-client->-r requirements.txt (line 5)) (1.25.8)
Requirement already satisfied: pytz>=2015.7 in /home/tobias/.local/lib/python3.8/site-packages (from babel>=1.3->sphinx->-r requirements.txt (line 1)) (2020.1)
Requirement already satisfied: pyparsing>=2.0.2 in /home/tobias/.local/lib/python3.8/site-packages (from packaging->sphinx->-r requirements.txt (line 1)) (2.4.7)
Requirement already satisfied: text-unidecode>=1.3 in /home/tobias/.local/lib/python3.8/site-packages (from python-slugify<5.0.0->transifex-client->-r requirements.txt (line 5)) (1.3)
Requirement already satisfied: gitdb<5,>=4.0.1 in /home/tobias/.local/lib/python3.8/site-packages (from gitpython<4.0.0->transifex-client->-r requirements.txt (line 5)) (4.0.5)
Requirement already satisfied: smmap<4,>=3.0.1 in /home/tobias/.local/lib/python3.8/site-packages (from gitdb<5,>=4.0.1->gitpython<4.0.0->transifex-client->-r requirements.txt (line 5)) (3.0.4)`

[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst]: Maybe your branch is too old. Run `git fetch upstream` and then `git rebase upstream/manual-2.3.x`. Then reinstall the dependencies.

PS: Please use 3 backticks for starting/ending code blocks. I edited your comment.

File: source/hardware/controllers/denon_mc7000.rst [hunk 7]
```diff
@@ -439,13 +440,13 @@ Button LEDs are fully mapped for the first function. As you press and hold :hwla
 
 Platter Ring LEDs are correlated with the :hwlabel:`VINYL` button.
 
-    -  If VINYL Mode is set ON then the LED follows the 33.3 rpm value.
-    -  If VINYL Mode is set OFF then the current track position is indicated by the Platter LEDs starting at the top.
+  - If VINYL Mode is set ON then the LED follows the 33.3 rpm value.
+  - If VINYL Mode is set OFF then the current track position is indicated by the Platter LEDs starting at the top.
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:444]:
```suggestion
  - If vinyl mode is enabled, the LED imitates a turntable platter spinning at 33.3 RPM.
  - If vinyl mode is disabled, the current track position is indicated by the Platter LEDs starting at the top.
```

(Or use `hwlabel` for VINYL if you want tot keep it uppercase.

**Review by Holzhaus — CHANGES_REQUESTED** [review]

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Please rebase on latest `manual-2.3.x`.

**Review by toszlanyi — COMMENTED** [review]
> Please rebase on latest `manual-2.3.x`.

I already started in `manual-2.3.x` - also the PR is related on that branch.

**Comment by toszlanyi** [comment]
I am really struggling with this site here. Where are all the commits coming from ? Why is there always an error show but I cannot find any ? Is there a way to check the rst format before commit? Thanks

**Comment by Holzhaus** [comment]
> I am really struggling with this site here. Where are all the commits coming from ? Why is there always an error show but I cannot find any ? Is there a way to check the rst format before commit? Thanks

`pre-commit install`

**Comment by toszlanyi** [comment]
Yay !!! All checks passed.
@Holzhaus - thanks a lot for your patience, guidance and to change branches back and forth to clean this up again.
Hope all is good now. Cheers!

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:81]: Hmm, I checked this branch out locally and removed them, to compare how it looks. IMHO, it's better without the lines. The main issue is that this is a very long list that is hard to navigate. How about splitting up the list of controls into different sections, like the Pioneer DDJ-200 mapping does?

https://manual.mixxx.org/2.3/en/hardware/controllers/pioneer_ddj_200.html

Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:81]: > How about splitting up the list of controls into different sections, like the Pioneer DDJ-200 mapping does?

Could do so but a re-write will take me some time as I am busy on other things right now.

File: source/hardware/controllers/denon_mc7000.rst [hunk 8]
```diff
@@ -102,301 +78,381 @@ Layout and functions
...
+    **Left Deck: Mix EQ** Turn these knobs to boost or cut the :hwlabel:`HI` and :hwlabel:`LOW` frequencies for Mic 1.
 
-12. **Sampler Volume:** Turn this knob to adjust the volume level of the
-    software’s sampler. If the volume is set to 0 then the Sampler banks
-    are hidden, otherwise shown.
+    **Right Deck:** :hwlabel:`TONE` Turn this knob to adjust the tone (high-low) of the Mic 2 audio signal.
 
-13. **Master Level Meters:** These LEDs show the Master Audio signal
-    (hardware). **They will not match with the Master Level in MIXXX.**
+    **Right Deck:** :hwlabel:`ECHO (MIC 1&2)` Turn this knob to adjust the amount of the echo effect on the microphones’ audio signal. This works on both: Mic 1 and Mic 2.
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:182]: Oh, don't worry, I'll fix the `hwlabel` implementation.

[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst:182]: Fixes in #268.

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: > Please merge: [toszlanyi#1](https://github.com/toszlanyi/manual/pull/1)

Looks nice, not really shorter though but segments give a better overview - still needs a bit of tuning. Thanks a lot for your input on this and let me finish until Tuesday hopefully. Then I send another commit to this PR. Cheers!

File: source/hardware/controllers/denon_mc7000.rst [hunk 9]
```diff
@@ -77,326 +50,216 @@ thread.
...
+15.       :hwlabel:`BOOTH LEVEL` knob                                      Turn to adjust the volume level of the booth outputs.
+20.       :hwlabel:`PHONES LEVEL` knob                                     Turn to adjust the volume level of the headphones.
+21.       :hwlabel:`PHONES MIX` knob                                       Adjusts the software’s audio output to the headphones, mixing between the cue output and the master mix output.
+22.       :hwlabel:`SPLIT` switch                                          When this switch is in the :hwlabel:`ON` position, the headphone audio will be “split” such that all channels sent to cue channel are summed to mono and sent to the left headphone channel and the master mix is summed to mono and sent to the right channel. When the switch is in the Off position, the cue channel and master mix will be “blended” together.
+          :hwlabel:`X FADER CONTOUR`                                       Turn to ajust the slope of the crossfader curve, to the left for a smooth fade (mixing) or to the right for a sharp cut (scratching.
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:89]:
```suggestion
          :hwlabel:`X FADER CONTOUR`                                       Turn to adjust the slope of the crossfader curve, to the left for a smooth fade (mixing) or to the right for a sharp cut (scratching.
```

File: source/hardware/controllers/denon_mc7000.rst [hunk 10]
```diff
@@ -77,326 +50,216 @@ thread.
...
+========  ===============================================================  ===============================================================
+
+.. note::
+   Input source selection, the :hwlabel:`LEVEL` meter, the :hwlabel:`MASTER LEVEL`/:hwlabel:`BOOTH LEVEL`/:hwlabel:`PHONES LEVEL`/:hwlabel:`PHONES MIX` knobs and the :hwlabel:`SPLIT` button are handled by the controller.
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:93]: I just checked on my phone, this line isn't wrapped properly. Maybe this will help:

```suggestion
   Input source selection, the :hwlabel:`LEVEL` meter, the :hwlabel:`MASTER LEVEL`, :hwlabel:`BOOTH LEVEL`, :hwlabel:`PHONES LEVEL` and :hwlabel:`PHONES MIX` knobs and the :hwlabel:`SPLIT` button are handled by the controller.
```

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (Holzhaus) [reply source/hardware/controllers/denon_mc7000.rst]: You can add some `=` to the rightmost header. That will make the right column wider in comparison to the other two columns.

File: source/hardware/controllers/denon_mc7000.rst (hunk 9 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:89]: I recognized that but also another spelling issue. Don't worry about that. Cheers

File: source/hardware/controllers/denon_mc7000.rst (hunk 6 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: I like the alternating row colors but the more I use the table view the less I like it. Unfortunately the table wont get wider in the rendered site adding `=`. Also on your DJ505 site I don't like the behavior of the table that you need to move the scrollbar to the right in order to see more than 6 words. This becomes worst in case the table is longer than the page height as you need to read first 6 words then scroll down, scroll right, scroll up again to find the initial point and read the last part of the description while the first part is invisible in column 3. Maybe you scrolled too far and then you only catch the first few words and the last but not the middle of the sentence. Really annoying. It would help to define an automatic line break for columns so that longer lines break into the next one into the same column. This makes it less annoying but increases the length of the document even more.

**Comment by toszlanyi** [comment]
I have now changed the whole document using CSV tables. This way the column width inside each table can be easily adjusted and imo it is easier writing the rst document

File: source/hardware/controllers/denon_mc7000.rst [hunk 11]
```diff
@@ -77,326 +50,253 @@ thread.
...
+   All functions are handled by the controller in hardware.
+   The status of Mixxx will not be changed.
+
+
+Deck Section
+~~~~~~~~~~~~
+
+.. csv-table::
+   :header: "No.", "Control", "Function"
+   :widths: 10 90 250
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:123]: This is wrong, the sum has to be 100%.

File: source/hardware/controllers/denon_mc7000.rst [hunk 12]
```diff
@@ -1,18 +1,7 @@
 Denon MC7000
-============
+------------
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:2]: Why did you change the header underlines? This broke the document tree:

Screenshot from 2020-11-03 12-46-04

File: source/hardware/controllers/denon_mc7000.rst [hunk 13]
```diff
@@ -77,326 +50,253 @@ thread.
...
+
+
+Mixer Section
+~~~~~~~~~~~~~
+
+.. csv-table::
+   :header: "No.", "Control", "Function"
+   :widths: 10 50 250
+
+   "1.", "Input Selector Switch", "Set to the desired audio source from this channel. :hwlabel:`LINE/PHONO` will disable the corresponding Deck (switches off all LEDs and the controller will not send MIDI signals to control Mixxx). This allows the mixer to use an external source. :hwlabel:`USB` will turn the corresponding Deck on and allows to control Mixxx via USB connection."
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:73]: This is very hard to read in plain text IMHO.

**Review by Holzhaus — CHANGES_REQUESTED** [review]
The CSV tables are very hard to read and edit in plain text IMHO, why don't you just use the simple tables instead?

If you want to make column 2 twice as wide as column 1, just use the appropriate number of `=`:

```rst
======  ============
Col 1   Col 2
======  ============
Row 1a  Row 1b
Row 2a  Row 2b
======  ============
```

File: source/hardware/controllers/denon_mc7000.rst (hunk 11 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:123]: Not necessarily. The default is 100% but any values are possible to define a ratio between the column width.

File: source/hardware/controllers/denon_mc7000.rst (hunk 12 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:2]: Never got there to modify anything. So when it changes this was not by purpose. Will correct it.

File: source/hardware/controllers/denon_mc7000.rst (hunk 13 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:73]: yep. inside tables it is hard to format. This was better in my previous non-tabled view.

**Comment by toszlanyi** [comment]
> The CSV tables are very hard to read and edit in plain text IMHO, why don't you just use the simple tables instead?
> If you want to make column 2 twice as wide as column 1, just use the appropriate number of `=`:

No! I found it much harder to read when the table was formatted as in your PR so I did re-write. For me this table is much better to read in source file.

And for the formatting it is terrible using `=` in 3 lines and then sometimes the text for column 2 also needed to be aligned. For PAD modes the 2nd column was very wide and 3rd only narrow even I extended until like this:
`=====  =====  ==========================================`

Now, with the csv-table I just change one number and the whole column width is formatted accordingly. Much, much easier.

**Comment by Holzhaus** [comment]
> No! I found it much harder to read when the table was formatted as in your PR so I did re-write. For me this table is much better to read in source file.

Are you really arguing that the table in the first screenshot is easier to read than the one in the second screenshot?

Screenshot from 2020-11-03 16-55-10

Screenshot from 2020-11-03 16-56-28

> And for the formatting it is terrible using = in 3 lines and then sometimes the text for column 2 also needed to be aligned, otherwise I got erromessages.

If your text editor is not up to the task, I recommend a powerful editor like `vim` (block-visual mode) or use a plugin (e.g. this one for [Atom](https://atom.io/packages/table-editor)).

**Comment by toszlanyi** [comment]
> > No! I found it much harder to read when the table was formatted as in your PR so I did re-write. For me this table is much better to read in source file.
> 
> Are you really arguing that the table in the first screenshot is easier to read than the one in the second screenshot?
> 
> > And for the formatting it is terrible using = in 3 lines and then sometimes the text for column 2 also needed to be aligned, otherwise I got erromessages.
> 
> If your text editor is not up to the task, I recommend a powerful editor like `vim` (block-visual mode) or use a plugin (e.g. this one for [Atom](https://atom.io/packages/table-editor)).

We don't need to argue. As a picture it looks more structured of course, but finding the lines far away is not nice. But thanks for the hint. I am using KDevelop and changed color theme to "vim dark". That at least selects the whole line when hovering the mouse over it. Anyhow, I like reading lines one by one in a row. But for you I can make it a nice picture like first table here:

grafik

File: source/hardware/controllers/denon_mc7000.rst (hunk 13 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: I found a way that looks quite okay in the rendered page - even though the source looks a bit strange. But as the result is better we should keep it like this.

**Comment by toszlanyi** [comment]
Last commit for this one. I struggle again on my depression for like 2 weeks. Get easily frustrated even with small changes and it is so hard to even start small tasks. I think we have got a good status already. Thanks a lot for all your involvement. Cheers

**Comment by Holzhaus** [comment]
> Last commit for this one. I struggle again on my depression for like 2 weeks. Get easily frustrated even with small changes and it is so hard to even start small tasks.

Sorry to hear that, I hope you'll get better soon!

I think we can merge this as-is, since it's a nice improvement even though I'm still not a fan of the CSV tables and would prefer the simple ones :wink:

File: source/hardware/controllers/denon_mc7000.rst [hunk 14]
```diff
@@ -21,51 +10,35 @@ analogue (vinyl) and digital sources.
 
 .. versionadded:: 2.2.4
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:11]: Could you please describe the functional mapping changes in a few words?

```suggestion
.. versionadded:: 2.2.4
.. versionchanged:: 2.3.0
   Please describe the changes in a few words here.
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Sorry, I missed that the mapping is already part of Mixxx 2.2.4 and these changes go into 2.3. Last thing to take care of for now, then we can merge.

File: source/hardware/controllers/denon_mc7000.rst (hunk 14 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:11]: i have now updated the changelog. I have used :hwlabel: initially but that looked like a mess. I'd prefer to keep it simple without :hwlabel: inside the changelog. Thanks a lot!

**Comment by toszlanyi** [comment]
No point to be sorry - struggle for years so it is known to me. Working on a structured way, like this mapping and documentation is good for my brain especially when things going bad the structure in math, physics and here is helpful. But if it comes worst then there is no chance anymore. Now Winter coming, Girlfriend in hospital, Lockdown again so my brain going mad.

About csv tables - the more i used them the more i liked it. The `=` is a mess for later formatting.

File: source/hardware/controllers/denon_mc7000.rst [hunk 15]
```diff
@@ -21,51 +10,51 @@ analogue (vinyl) and digital sources.
...
+.. versionchanged:: 2.3.0
+
+-  Bugfix: fix Vinylmode for Deck 4
+-  Feature: add mapping to sort library for BPM, Key, Artist and Title
+-  Feature: add Softtakeover for Pitch Faders
+-  Feature: add to eject track from Deck with Library Select knob
+-  Feature: add to search through track using Jog Wheel
+-  Feature: add Waveform zoom and reset to default
+-  Feature: change Parameter buttons to select star rating and color for tracks
+-  Feature: change of Censor and Reverse as well as Backspin access
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:25]: Let's not make a full changelog, a few words suffice. 

```suggestion
Fixed vinyl mode for deck 4 and added various new features such as library sorting, jog wheel strip search, waveform zoom, censoring, backspinning, track colors, etc. 
```

File: source/hardware/controllers/denon_mc7000.rst [hunk 16]
```diff
@@ -21,51 +10,43 @@ analogue (vinyl) and digital sources.
...
 .. versionadded:: 2.2.4
 
-Features
-~~~~~~~~
+-  Initial mapping release
+
+
+.. versionchanged:: 2.3.0
+
+-  Fixed a bug for Vinylmode on Deck 4 and added new features, like library sorting, search through track with Jog Wheel, eject track from Deck, added waveform zoom, changed Censor/Reverse/Spinback and Parameter button mapping
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:18]: It's still an enumeration, which breaks the markup ;-)

```suggestion
   Fixed vinyl mode on deck 4 and added new features, like library sorting, search through track with Jog Wheel, eject track from deck, added waveform zoom, changed censor/reverse/spinback and parameter button mapping.
```

**Comment by toszlanyi** [comment]
all fixed now I hope. Have learned much more about reST fomating. Just the tables for Pad buttons, especially for the Roll Mode wont align centered. I tried even that supposed in on area...

```
+-----+-----+-----+-----+
| 1/16| 1/8 | 1/4 | 1/2 |
+-----+-----+-----+-----+
||  1 ||  2 ||  4 ||  8 |
+-----+-----+-----+-----+
```
but always left aligned inside cells. otherwise i am really happy with the look of that site now

File: source/hardware/controllers/denon_mc7000.rst [hunk 17]
```diff
@@ -21,51 +10,43 @@ analogue (vinyl) and digital sources.
...
 .. versionadded:: 2.2.4
 
-Features
-~~~~~~~~
+Initial mapping release
+
+
+.. versionchanged:: 2.3.0
+
+Fixed a bug for Vinylmode on Deck 4 and added new features, like library sorting, search through track with Jog Wheel, eject track from Deck, added waveform zoom, changed Parameter and Censor/Reverse/Spinback button mapping
```
Comment (Holzhaus) [inline source/hardware/controllers/denon_mc7000.rst:18]: Markup is still broken. This needs to be indented by 3 spaces and without a blank line between `versionadded` and the text block. I'll just merge this and fix this myself :D

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you.

File: source/hardware/controllers/denon_mc7000.rst (hunk 17 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:18]: sorry for this - obviously i didnt understand how you wanna have this formatted
